app/views/layouts/application.html.haml

```diff
@@ -16,7 +16,7 @@
 
     = csrf_meta_tags
 
-  %body.no-js
+  %body.no-js{ 'class': "#{params[:controller]} #{params[:action]}"}
```

Comment:
What about a single class that is the controller and action concatenated e.g. "dashboard-faq" rather than "dashboard faq"? There are plenty of routes where the action is just "show" so it seems odd to have lots of different pages all with a body class of "show".